### PR TITLE
feat: adds Superset usage metadata charts to Operator Dashboard

### PR DESCRIPTION
**Pull request by pomegranited**
Adds a Superset tab to the Operator Dashboard with some introspection charts adapted from https://github.com/hometogo/hometogo-data-code-snippets. (I omitted charts tracking "naming conventions" and charts counting datasets over time, as these didn't seem interesting to the Aspects project.)

Closes: https://github.com/openedx/tutor-contrib-aspects/issues/427

**Screenshots**

superset_tab

**Author Notes & Concerns**

- [x] Fix filter warning shown when dashboard loads ("Error loading chart datasources; Filters may not work correctly.") 8d71dfc9b9ad2cdf3968310af6ac2b4427e06563
- [x] Add chart descriptions f0db2893f6c53e48056190c62907a786eab972bc
- [ ] More charts?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: tutoraspects/templates/openedx-assets/assets/datasets/monitoring.superset_action_log.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+_file_name: monitoring.superset_action_log.yaml
+table_name: superset_action_log
+main_dttm_col: null
+description: null
+default_endpoint: null
+offset: 0
+cache_timeout: null
+schema: {{SUPERSET_DB_METADATA_NAME}}
+sql: "select\nl.dttm as action_date,\ncase \n  when lower(l.action)\
```
Comment (pomegranited) [inline tutoraspects/templates/openedx-assets/assets/datasets/monitoring.superset_action_log.yaml:9]: note: [original query](https://github.com/hometogo/hometogo-data-code-snippets/blob/0db5a76aa9a814a93223e2932ed081a7750e80c9/superset/usage_dashboard/dashboard_export/datasets/SuperSet_Metadata/monitoring.superset_action_log.yaml#L8) used PostgreSQL's `date_trunc('day', l.dttm) as action_date` here.

MySQL's equivalent is `extract(day from l.dttm) as action_date`, but this caused errors in the charts, so I left the `l.dttm` datetime field as-is.

**Review by Ian2012 — APPROVED** [review]

**Comment by Ian2012** [comment]
The CI is failing because the final `assets.yaml` is too big. An contribution is expected from our team, I'll let you know to refactor it

File: tutoraspects/templates/aspects/jobs/init/init-mysql.sh [hunk 2]
```diff
@@ -16,6 +16,13 @@ echo "MySQL is up and running"
...
 mysql -u {{ MYSQL_ROOT_USERNAME }} --password="{{ MYSQL_ROOT_PASSWORD }}" --host "{{ MYSQL_HOST }}" --port {{ MYSQL_PORT }} -e "ALTER USER '{{ SUPERSET_DB_USERNAME }}'@'%' IDENTIFIED BY '{{ SUPERSET_DB_PASSWORD }}';"
 mysql -u {{ MYSQL_ROOT_USERNAME }} --password="{{ MYSQL_ROOT_PASSWORD }}" --host "{{ MYSQL_HOST }}" --port {{ MYSQL_PORT }} -e "GRANT ALL ON {{ SUPERSET_DB_NAME }}.* TO '{{ SUPERSET_DB_USERNAME }}'@'%';"
+
+# Superset metadata user (read-only)
+mysql -u {{ MYSQL_ROOT_USERNAME }} --password="{{ MYSQL_ROOT_PASSWORD }}" --host "{{ MYSQL_HOST }}" --port {{ MYSQL_PORT }} -e "CREATE USER IF NOT EXISTS '{{ SUPERSET_DB_METADATA_USERNAME }}';"
+mysql -u {{ MYSQL_ROOT_USERNAME }} --password="{{ MYSQL_ROOT_PASSWORD }}" --host "{{ MYSQL_HOST }}" --port {{ MYSQL_PORT }} -e "ALTER USER '{{ SUPERSET_DB_METADATA_USERNAME }}'@'%' IDENTIFIED BY '{{ SUPERSET_DB_METADATA_PASSWORD }}';"
+mysql -u {{ MYSQL_ROOT_USERNAME }} --password="{{ MYSQL_ROOT_PASSWORD }}" --host "{{ MYSQL_HOST }}" --port {{ MYSQL_PORT }} -e "GRANT SELECT, EXECUTE ON {{ SUPERSET_DB_METADATA_NAME }}.* TO '{{ SUPERSET_DB_METADATA_USERNAME }}'@'%';"
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/jobs/init/init-mysql.sh:28]: Can we limit this to the tables we need? I'm somewhat concerned about things like access to the `dbs` table which stores credentials in plaintext.

**Review by bmtcril — CHANGES_REQUESTED** [review]
Looks good, thanks for taking this on! Just requesting the one change.

**Comment by bmtcril** [comment]
@Ian2012 do you have a ticket or an ETA for the configmap changes? I'm likely going to run into this myself in my next PR.

**Comment by Ian2012** [comment]
@bmtcril not sure, the plan is to build the image with the assets.yml file bundled in it and extend it via configmap

**Comment by pomegranited** [comment]
I've addressed your review @bmtcril .  I can work on adding Clickhouse performance metrics next week, or as part of a separate PR, if you'd prefer.

**Comment by bmtcril** [comment]
I think a separate pr would be good. That could probably all go in the existing ClickHouse tab in the operator dash.

**Comment by pomegranited** [comment]
@bmtcril 

> I think a separate pr would be good. That could probably all go in the existing ClickHouse tab in the operator dash.

👍 

Should these Superset charts go in the Operator Dashboard too? I can easily move the tabs over there if that's better.

**Comment by bmtcril** [comment]
@pomegranited If you can put these in the operator dash that would be great, it should help with keeping the permissions easy. Otherwise I think once you get past the current errors and rebase, this should be good to go.

**Comment by pomegranited** [comment]
Ready for another review @bmtcril @Ian2012 . I've:
* moved the superset metadata charts into a new Superset tab on the Operator Dashboard
* split the SQL `GRANT`s for the metadata tables into an init script that runs after Superset migrations (can't GRANT access to a table that doesn't exist)
* redeployed from scratch on my local tutor dev.

**Review by bmtcril — APPROVED** [review]
Looks great, thanks for all the hard work and iteration!

**Comment by openedx-webhooks** [comment]
@pomegranited 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/488&entry.1484655318=https://github.com/pomegranited&entry.752974735=2023-10-24+07:20&entry.1917517419=2023-11-02+12:42&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**